nixos/modules/security/pam.nix

```diff
@@ -1174,13 +1235,26 @@ let
                 control = "sufficient";
                 modulePath = "${config.systemd.package}/lib/security/pam_systemd_home.so";
               }
+              {
+                # before pam_unix.so
+                name = "pwquality";
+                enable = cfg.pwquality.enable;
+                control = cfg.pwquality.control;
+                modulePath = "${lib.getLib cfg.pwquality.package}/lib/security/pam_pwquality.so";
+                settings = cfg.pwquality.settings;
+              }
               {
                 name = "unix";
                 control = "sufficient";
                 modulePath = "${package}/lib/security/pam_unix.so";
                 settings = {
                   nullok = true;
                   yescrypt = true;
+                }
+                // lib.optionalAttrs cfg.pwquality.enable {
```

Comment:
You don't need to make the key optional. You can just set `use_authtok = cfg.pwquality.enable;` because `false` keys are not rendered.